native/src/main/java/io/ballerina/stdlib/http/transport/contractimpl/DefaultHttpResponseFuture.java

```diff
@@ -47,7 +47,8 @@
     private HttpConnectorListener pushPromiseListener;
     private ConcurrentHashMap<Integer, HttpConnectorListener> pushResponseListeners;
     private ConcurrentHashMap<Integer, Throwable> pushResponseListenerErrors;
-    private BackPressureHandler backPressureHandler;
+    private BackPressureHandler requestBackPressureHandler;
+    private BackPressureHandler responseBackPressureHandler;
```

Comment:
At any given time, there could be only one of these two. Because the same CarbonMessage can't be both outbound request and outbound response. Therefore, can't we just have one field as we did earlier ?

Reply:
Fixed via [`9b3e185`](https://github.com/ballerina-platform/module-ballerina-http/pull/852/commits/9b3e18564129f40f3c02abd9d85db3910db543d0)